src/presign/participant.rs

```diff
@@ -875,32 +853,14 @@ impl PresignParticipant {
 /// identifiers.
 pub(crate) fn get_keyshare(
     self_id: ParticipantIdentifier,
-    auxinfo_identifier: Identifier,
-    keyshare_identifier: Identifier,
-    storage: &Storage,
+    input: &Input,
 ) -> Result<PresignKeyShareAndInfo> {
     // Reconstruct keyshare from local storage
     let keyshare_and_info = PresignKeyShareAndInfo {
-        aux_info_private: storage.retrieve(
-            PersistentStorageType::AuxInfoPrivate,
-            auxinfo_identifier,
-            self_id,
-        )?,
-        aux_info_public: storage.retrieve(
-            PersistentStorageType::AuxInfoPublic,
-            auxinfo_identifier,
-            self_id,
-        )?,
-        keyshare_private: storage.retrieve(
-            PersistentStorageType::PrivateKeyshare,
-            keyshare_identifier,
-            self_id,
-        )?,
-        keyshare_public: storage.retrieve(
-            PersistentStorageType::PublicKeyshare,
-            keyshare_identifier,
-            self_id,
-        )?,
+        aux_info_private: input.auxinfo_private.clone(),
+        aux_info_public: input.find_auxinfo_public(self_id)?.clone(),
+        keyshare_private: input.keyshare_private.clone(),
+        keyshare_public: input.find_keyshare_public(self_id)?.clone(),
```

Comment:
thought: cleaning this type up a bit might be a good option for a separate clean-up ticket.